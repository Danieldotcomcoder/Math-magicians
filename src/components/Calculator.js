import React from 'react';
// import operate from '../logic/operate';
import calculate from '../logic/calculate';
// eslint-disable-next-line react/prefer-stateless-function
class Calculator extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      next: null,
      total: null,
    };
    this.handler = this.handler.bind(this);
  }

  handler(event) {
    this.setState((state) => calculate(state, event.target.innerText));
  }

  render() {
    const { next, total } = this.state;
    return (
      <div className="calculator">
        <div className="calc-output">
          <div className="calc-result">
            {next || (total || '0')}
          </div>
        </div>
        <div className="calc-grid">
          <button type="button" onClick={this.handler}> AC </button>
          <button type="button" onClick={this.handler}> +/- </button>
          <button type="button" onClick={this.handler}> % </button>
          <button type="button" className="logic-operators" onClick={this.handler}> รท </button>
          <button type="button" onClick={this.handler}>7</button>
          <button type="button" onClick={this.handler}> 8 </button>
          <button type="button" onClick={this.handler}> 9 </button>
          <button type="button" className="logic-operators" onClick={this.handler}> x </button>
          <button type="button" onClick={this.handler}> 4 </button>
          <button type="button" onClick={this.handler}> 5 </button>
          <button type="button" onClick={this.handler}> 6 </button>
          <button type="button" className="logic-operators" onClick={this.handler}> - </button>
          <button type="button" onClick={this.handler}> 1 </button>
          <button type="button" onClick={this.handler}> 2 </button>
          <button type="button" onClick={this.handler}> 3 </button>
          <button type="button" className="logic-operators" onClick={this.handler}> + </button>
          <button type="button" id="zero" onClick={this.handler}> 0 </button>
          <button type="button" onClick={this.handler}> . </button>
          <button type="button" className="logic-operators" onClick={this.handler}> = </button>
        </div>
      </div>

    );
  }
}

export default Calculator;
