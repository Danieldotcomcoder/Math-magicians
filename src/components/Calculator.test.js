import React from 'react';
import renderer from 'react-test-renderer';
import Calculator from './Calculator';

it('Renders Calculator page', () => {
  const calc = renderer.create(
    <Calculator />,
  ).toJSON();
  expect(calc).toMatchSnapshot();
});
