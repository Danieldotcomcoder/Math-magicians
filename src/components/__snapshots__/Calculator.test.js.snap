// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Calculator page 1`] = `
<div
  className="calc-page"
>
  <h3
    className="math-state"
  >
    Lets do some math!
  </h3>
  <div
    className="calculator"
  >
    <div
      className="calc-output"
    >
      <div
        className="calc-result"
        role="none"
      >
        0
      </div>
    </div>
    <div
      className="calc-grid"
    >
      <button
        onClick={[Function]}
        type="button"
      >
         AC 
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
         +/- 
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
         % 
      </button>
      <button
        className="logic-operators"
        onClick={[Function]}
        type="button"
      >
         รท 
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
         8 
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
         9 
      </button>
      <button
        className="logic-operators"
        onClick={[Function]}
        type="button"
      >
         x 
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
         4 
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
         5 
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
         6 
      </button>
      <button
        className="logic-operators"
        onClick={[Function]}
        type="button"
      >
         - 
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="logic-operators"
        onClick={[Function]}
        type="button"
      >
         + 
      </button>
      <button
        id="zero"
        onClick={[Function]}
        type="button"
      >
         0 
      </button>
      <button
        onClick={[Function]}
        type="button"
      >
         . 
      </button>
      <button
        className="logic-operators"
        onClick={[Function]}
        type="button"
      >
         = 
      </button>
    </div>
  </div>
</div>
`;
